arch/rv32i/Cargo.toml

```diff
@@ -7,3 +7,4 @@ edition = "2018"
 [dependencies]
 kernel = { path = "../../kernel" }
 tock_rt0 = { path = "../../libraries/tock-rt0" }
+riscvregs = { git = "https://github.com/DieracDelta/riscvregs.git", rev = "c03eb6c" }
```

Comment:
Unfortunately currently Tock doesn't support (i.e. Tock doesn't allow) external dependencies because we cannot track `unsafe` use in dependencies currently. So, to use a library like this it would either need to be included in the `/library` folder in Tock, or included as a module in the rv32i crate.

Reply:
Would it be reasonable to add this in as a git submodule under libraries?

Reply:
Ideally we'd avoid git submodules where possible (it's just a bigger hassle to develop with). Moving this into the repo would be great. You can technically do so without losing the git history with some git magic (`git filter-branch` i believe).